sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -58,11 +58,11 @@ object JDBCRDD extends Logging {
     val dialect = JdbcDialects.get(url)
     val conn: Connection = JdbcUtils.createConnectionFactory(url, properties)()
     try {
-      val statement = conn.prepareStatement(s"SELECT * FROM $table WHERE 1=0")
+      val statement = conn.prepareStatement(dialect.getTableExistsQuery(table))
```

Comment:
I see your point. Postgres and MySQL use `SELECT 1 FROM $table LIMIT 1`. I suppose they could be `SELECT * ...` ? The upside is that we don't need two separate queries. The downside is returning a row instead of 1 in these two DB's cases?

I'll go ahead and make the change you suggest for a look.